include/libtorrent/aux_/session_impl.hpp

```diff
@@ -1054,15 +1065,28 @@ namespace aux {
 				, error_code& ec
 				, int flags);
 
+			void send_udp_packet_hostname_listen(aux::session_listen_socket* sock
```

Comment:
the hostname overload is only used for http and socks5 proxies, right? I would have expected that case to not depend on a source NIC. Or is it possible to have multiple connections to the same proxy, via different NICs?